packages/web/src/pages/saved-page/SavedPageProvider.tsx

```diff
@@ -72,12 +85,33 @@ type SavedPageState = {
 class SavedPage extends PureComponent<SavedPageProps, SavedPageState> {
   state: SavedPageState = {
     filterText: '',
+    sortMethod: '',
+    sortDirection: '',
     initialOrder: null,
     currentTab: ProfileTabs.TRACKS
   }
 
+  handleFetchSavedTracks = debounce(() => {
+    this.props.fetchSavedTracks(
+      this.state.filterText,
+      this.state.sortMethod,
+      this.state.sortDirection
+    )
+  }, 300)
+
+  handleFetchMoreSavedTracks = (offset: number, limit: number) => {
+    const { filterText, sortMethod, sortDirection } = this.state
+    this.props.fetchMoreSavedTracks(
+      filterText,
+      sortMethod,
+      sortDirection,
+      offset,
+      limit
+    )
+  }
+
   componentDidMount() {
-    this.props.fetchSavedTracks()
+    this.handleFetchSavedTracks()
```

Comment:
nit, i think this should not call the same method w/ the debounce since this will only happen once. Can we debounce at the caller side?